src/MSBuild/TerminalLogger/NodeStatus.cs

```diff
@@ -13,13 +13,42 @@ internal class NodeStatus
 {
     public string Project { get; }
     public string? TargetFramework { get; }
+    public TerminalColor TargetPrefixColor { get; } = TerminalColor.Default;
+    public string? TargetPrefix { get; }
     public string Target { get; }
     public StopwatchAbstraction Stopwatch { get; }
 
+    /// <summary>
+    /// Status of a node that is currently doing work.
+    /// </summary>
+    /// <param name="project">The project that is written on left side.</param>
+    /// <param name="targetFramework">Target framework that is colorized and written on left side after project.</param>
+    /// <param name="target">The currently running work, usually the currently running target. Written on right.</param>
+    /// <param name="stopwatch">Duration of the current step. Written on right after target.</param>
     public NodeStatus(string project, string? targetFramework, string target, StopwatchAbstraction stopwatch)
+    {
+        Debug.Assert(!target.Contains("\x1B"), "Target should not contain any escape codes, if you want to colorize target use the other constructor.");
+        Project = project;
+        TargetFramework = targetFramework;
+        Target = target;
+        Stopwatch = stopwatch;
+    }
+
+    /// <summary>
+    /// Status of a node that is currently doing work.
+    /// </summary>
+    /// <param name="project">The project that is written on left side.</param>
+    /// <param name="targetFramework">Target framework that is colorized and written on left side after project.</param>
+    /// <param name="targetPrefixColor">Color for the status of the currently running work written on right.</param>
+    /// <param name="targetPrefix">Colorized status for the currently running work, written on right, before target, and separated by 1 space from it.</param>
+    /// <param name="target">The currently running work, usually the currently runnig target. Written on right.</param>
+    /// <param name="stopwatch">Duration of the current step. Written on right after target.</param>
+    public NodeStatus(string project, string? targetFramework, TerminalColor targetPrefixColor, string targetPrefix, string target, StopwatchAbstraction stopwatch)
```

Comment:
@rainersigwald  I've replaced the regex with passing the pieces directly, and coloring them later, this way I don't need to strip the ansi codes, or allocate additional structures to represent a string with colors. I did the bare minimum to not make the math in Render become a nightmare. 

I am not super happy with the 2 constructors, let me know if you'd prefer to have just this one, and replace the calls to the other one.

Reply:
NIT: Please consider calling the other constructor by  `: this()` . There already is that `if (target.Contains("\x1B"))` debug assertions which is missed in this constructor

Reply:
done